### PR TITLE
Add units and decimals to metrics dashboard

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

We needed decimals and units added to metrics. This does it!  
image

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by michielderoos** [comment]
[Ensure Decimals are only shown in top level statistics when appropriate](https://trello.com/c/kYJVnE3L/519-ensure-decimals-are-only-shown-in-top-level-statistics-when-appropriate)

**Comment by michielderoos** [comment]
[Make sure all top level statistics include appropriate units](https://trello.com/c/HJVPxemK/518-make-sure-all-top-level-statistics-include-appropriate-units)

**Comment by tristanhcole** [comment]
I think we should resolve #251 before implementing. Some currencies, such as Vatu in Vanuatu, don't use decimals, whereas others, like USD do.

File: app/client/components/dashboard/card/CustomTabs.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ import {
 const { TabPane } = Tabs;
 const { Text } = Typography;
 
-import { replaceUnderscores, toTitleCase } from "../../../utils";
+import { replaceUnderscores, toTitleCase, toDollars } from "../../../utils";
```
Comment (tristanhcole) [inline app/client/components/dashboard/card/CustomTabs.jsx:18]: I'm either going crazy or `toDollars` doesn't seem to be defined in `utils.js`.

File: app/client/components/dashboard/card/GroupByChart.jsx [hunk 2]
```diff
@@ -20,7 +20,10 @@ class GroupByChart extends React.Component {
 
     const aggregate = this.props.data.aggregate;
     const aggregateKeys = aggregate ? Object.keys(aggregate) : [];
-    const aggregateData = aggregate ? Object.values(aggregate) : [];
+    var aggregateData = aggregate ? Object.values(aggregate) : [];
+    if (this.props.data.type.type == "currency") {
```
Comment (tristanhcole) [inline app/client/components/dashboard/card/GroupByChart.jsx:24]: 1) `this.props.data.type.type` seems a bit messy. Can we be more explicit?
2) don't love the string comparison, but not sure if there is a neater way to do it

File: app/client/components/dashboard/card/CustomTabs.jsx (hunk 1 above)
Comment (tristanhcole) [reply app/client/components/dashboard/card/CustomTabs.jsx:18]: Also `toDollars` implies we will always be using "dollars"... should this be `toCurrency`?

File: app/server/utils/metrics/metric.py [hunk 3]
```diff
@@ -66,6 +66,16 @@ def execute_query(self, user_filters: dict = None, date_filters_dict=None, enabl
 
         if self.is_timeseries:
             result = {}
+            if self.value_type not in VALUE_TYPES:
+                raise Exception(f'{self.value_type} not a valid metric type!')
+            result['type'] = {
+                'type': self.value_type,
+                'display_decimals': 2 if self.value_type == COUNT_AVERAGE else 0
```
Comment (tristanhcole) [inline app/server/utils/metrics/metric.py:73]: `change this to self.token.display_decimals once #251 is merged` also?

File: app/server/utils/metrics/metric.py [hunk 4]
```diff
@@ -66,6 +66,16 @@ def execute_query(self, user_filters: dict = None, date_filters_dict=None, enabl
...
+            }
+            if self.token:
+                result['type']['currency_name'] = self.token.name
+                result['type']['currency_symbol'] = self.token.symbol
+                result['type']['display_decimals'] = 2 # change this to self.token.display_decimals once #251 is merged
```
Comment (tristanhcole) [inline app/server/utils/metrics/metric.py:78]: 👍

File: app/server/utils/metrics/metric.py [hunk 5]
```diff
@@ -66,6 +66,16 @@ def execute_query(self, user_filters: dict = None, date_filters_dict=None, enabl
 
         if self.is_timeseries:
             result = {}
+            if self.value_type not in VALUE_TYPES:
+                raise Exception(f'{self.value_type} not a valid metric type!')
+            result['type'] = {
+                'type': self.value_type,
```
Comment (tristanhcole) [inline app/server/utils/metrics/metric.py:72]: can we use something more explicit than type? `value_type`?

File: app/server/utils/metrics/metrics_const.py [hunk 6]
```diff
@@ -71,3 +71,9 @@
 
 # Misc Values
 GROUPED = 'grouped'
+
+# Metric types
+COUNT = 'count'
+COUNT_AVERAGE = 'count_average'
+CURRENCY = 'currency'
+VALUE_TYPES = [COUNT, COUNT_AVERAGE, CURRENCY]
```
Comment (tristanhcole) [inline app/server/utils/metrics/metrics_const.py:79]: 👍

Comment (tristanhcole) [reply app/server/utils/metrics/metrics_const.py:79]: don't love string comparison on frontend but not sure of a better alternative

File: app/server/utils/metrics/metric.py (hunk 4 above)
Comment (tristanhcole) [reply app/server/utils/metrics/metric.py:78]: For the tests, could we have a token with `display_tokens=0`  and `display_tokens=2`. We'll have both these usecases live with USD and VT (vatu)

**Review by tristanhcole — CHANGES_REQUESTED** [review]
A couple of small changes/questions, next review should be a quick turnaround! 🧮

File: app/client/components/dashboard/card/CustomTabs.jsx (hunk 1 above)
Comment (michielderoos) [reply app/client/components/dashboard/card/CustomTabs.jsx:18]: D'oh! Forgot to add that change to the PR. Good catch!

File: app/server/utils/metrics/metric.py (hunk 3 above)
Comment (michielderoos) [reply app/server/utils/metrics/metric.py:73]: Yep! Added according comment

File: app/server/utils/metrics/metrics_const.py (hunk 6 above)
Comment (michielderoos) [reply app/server/utils/metrics/metrics_const.py:79]: Still a string comparison, but I added PAYMENT to `constants.js` so it's a bit cleaner now

File: app/client/components/dashboard/card/GroupByChart.jsx (hunk 2 above)
Comment (michielderoos) [reply app/client/components/dashboard/card/GroupByChart.jsx]: 1. Changed to `value_type`  
2. Made it a const, so it's a little bit cleaner now

File: app/server/utils/metrics/metric.py (hunk 3 above)
Comment (michielderoos) [reply app/server/utils/metrics/metric.py:74]: This is for `COUNT_AVERAGE` (so like Volume Per User). It's set to `token.display_decimals` on l78 for anything involving a token

File: app/server/utils/metrics/metric.py (hunk 4 above)
Comment (michielderoos) [reply app/server/utils/metrics/metric.py]: Added a test! ✅

**Review by tristanhcole — APPROVED** [review]
Tested. Looks good to me! 🌟

**Pull request merged**